public/src/components/switchboard.tsx

```diff
@@ -1,456 +1,211 @@
-  state: Switches;
-  previousStateFromServer: DataFromServer | null;
-
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      oneOffPaymentMethods: {
-        stripe: SwitchState.Off,
-        stripeApplePay: SwitchState.Off,
-        stripePaymentRequestButton: SwitchState.Off,
-        payPal: SwitchState.Off,
-        amazonPay: SwitchState.Off,
-      },
-      recurringPaymentMethods: {
-        stripe: SwitchState.Off,
-        stripeApplePay: SwitchState.Off,
-        stripePaymentRequestButton: SwitchState.Off,
-        payPal: SwitchState.Off,
-        directDebit: SwitchState.Off,
-        existingCard: SwitchState.Off,
-        existingDirectDebit: SwitchState.Off,
-        sepa: SwitchState.Off,
-      },
-      enableDigitalSubGifting: SwitchState.Off,
-      useDotcomContactPage: SwitchState.Off,
-      enableRecaptchaBackend: SwitchState.Off,
-      enableRecaptchaFrontend: SwitchState.Off,
-      enableContributionsCampaign: SwitchState.Off,
-      forceContributionsCampaign: SwitchState.Off,
-      enableQuantumMetric: SwitchState.Off,
-      experiments: {},
-    };
-    this.previousStateFromServer = null;
-  }
-
-  UNSAFE_componentWillMount(): void {
-    this.fetchStateFromServer();
-  }
-
-  fetchStateFromServer(): void {
-    fetchSupportFrontendSettings(SupportFrontendSettingsType.switches).then(serverData => {
-      this.previousStateFromServer = serverData;
-      this.setState({
-        ...serverData.value,
-      });
-    });
-  }
-
-  updateOneOffPaymentMethodSwitch(paymentMethod: string, switchState: SwitchState): void {
-    this.setState(prevState =>
-      update(prevState, {
-        oneOffPaymentMethods: {
-          [paymentMethod]: { $set: switchState },
-        },
-      }),
-    );
-  }
-
-  updateRecurringPaymentMethodSwitch(paymentMethod: string, switchState: SwitchState): void {
-    this.setState(prevState =>
-      update(prevState, {
-        recurringPaymentMethods: {
-          [paymentMethod]: { $set: switchState },
-        },
-      }),
-    );
-  }
-
-  updateFeatureSwitch(switchName: string, switchState: SwitchState): void {
-    this.setState(prevState =>
-      update(prevState, {
-        experiments: {
-          [switchName]: {
-            state: { $set: switchState },
-          },
-        },
-      }),
+    updatedState[groupId].switches[mySwitchId].state = event.target.checked
+      ? SwitchState.On
+      : SwitchState.Off;
+
+    setData(updatedState);
+    setNeedToSaveDataWarning(true);
+  };
+
+  const createSwitchesFromGroupData = (
+    mySwitch: [string, Switch],
+    groupId: string,
+  ): JSX.Element => {
+    const [switchId, switchData] = mySwitch;
+
+    return (
+      <FormControlLabel
+        label={switchData.description}
+        checked={switchData.state === 'On' ? true : false}
+        control={<SwitchUI />}
+        onChange={(event): void => updateSwitchSetting(groupId, mySwitch, event)}
```

Comment:
you can avoid the `event: any` param in `updateSwitchSetting` if you just pass in a boolean here:
```
const checked = switchData.state === 'On';

return (
  <FormControlLabel
    label={switchData.description}
    checked={checked}
    control={<SwitchUI />}
    onChange={(event): void => updateSwitchSetting(groupId, mySwitch, !checked)}
    value={switchId}
    key={switchId}
  />
);
```

Reply:
Done and pushed fix